### DISCUSSION TITLE
Support reactions and Kudos

### DISCUSSION THREAD

**Discussion by dacog** [umputun/remark42#1784] [Ideas]
Hi,

I would like to implement reactions (and kudos) buttons into remark42.  

Are there any plans for this? 

If there are no plans, would you consider a merge request should I develop this? 

I have considered 2 ways to implement this, although I still don't know if they are feasible.  

1. Extend the database model and add reactions to it. Then expand the frontend to support the new reactions.  
2. Extend the frontend and implement reactions as some kind of comment using the api, which also gets a special "reactions" widget before the comments.  This way the r reactions could also show in the comments history and use the whole auth system.  Maybe this also requires a change in the database to add a comment type, which by default is unpopulated (so that it does not break anything).  

In both cases the reaction would be font icons.

What do you think? Would this be something interesting for the project or is this completely outside the scope of the project? 

Do you have any preferences or comments for the implementation?

Best,

Diego

**Comment by umputun** [comment]
As of now, remark42 supports two "reactions" with up and down arrows for voting, similar to the commonly used 👍 and 👎 emojis. Do we really need anything else? If so, what will the complete set be?

I'm generally not a big fan of reactions in comments. However, occasionally adding a 😄 reaction or something similar could be useful. So, I don't strongly object but would like to hear what other contributors think.

Technically, if we decide to go ahead with this idea, it should be a model extension on the backend side, not a special type of comment handled by the frontend. It's somewhat similar to the voting feature but with a map of [reaction_type]int (I assume you meant a predefined set of reactions, similar to GitHub). After the backend change, support for reactions should also be added to the frontend, and all of this needs to be fully backward-compatible.

@paskal @akellbl4 @Mavrin  - what do you think?

**Comment by dacog** [comment]
Hi @umputun 

Thank you for your reply.   It seems I didn't explain the idea clearly.  The idea here is to support reactions for the content (post, article) and not for the comments.  I'm aware that the comments support voting and I think thumbs up and down are ok for now.  

This would be something like the reactions in Disqus, where you don't need to comment but you can react to a piece of content or url.

**Comment by umputun** [comment]
I see, thanks for clarification
